### PR TITLE
Feat/authentication

### PR DESCRIPTION
**Pull request by karljerico**
### OVERVIEW

Add a working registration. Created a temporary homepage. Also initial login authentication without email verification. 

### NOTES

- Created a register user
- Edit login controller to authenticate 
- Initial home page and logout
- Connection of the login, register, and home page

### SNIPPETS

image
image

### CODE REVIEW

File: src/app/Http/Controllers/LoginController.php [hunk 1]
```diff
@@ -3,11 +3,31 @@
...
 {
     public function index(): View
     {
         return view('auth/login');
     }
+
+    public function login(Request $request)
+    {
+        $request->validate([
+            'loginname' => 'required',
```
Comment (hipe-francis) [inline src/app/Http/Controllers/LoginController.php]: Can you please moved this in a dedicated Form Request class? 🙇‍♂️

File: src/routes/web.php [hunk 2]
```diff
@@ -1,9 +1,11 @@
 <?php
 
-use App\Http\Controllers\RegisterController;
 use Illuminate\Support\Facades\Route;
+use App\Http\Controllers\HomeController;
 use App\Http\Controllers\LoginController;
-use App\Http\Controllers\UserController;
+use App\Http\Controllers\RegisterController;
+=======
```
Comment (hipe-francis) [inline src/routes/web.php]: please remove this

File: src/app/Http/Controllers/LoginController.php (hunk 1 above)
Comment (hipe01mark) [reply src/app/Http/Controllers/LoginController.php]: +1

Comment (karljerico) [reply src/app/Http/Controllers/LoginController.php]: I already resolved this comment, thank you sir!

File: src/routes/web.php (hunk 2 above)
Comment (karljerico) [reply src/routes/web.php]: I resolved this comment sir thank you!

File: src/routes/web.php [hunk 3]
```diff
@@ -17,7 +17,9 @@
...
 
 
-Route::get('/', [LoginController::class, 'index'])->name('login');
+Route::get('/', [LoginController::class, 'index'])->name('login.auth');
+Route::post('/', [LoginController::class, 'auth']);
 Route::get('/register', [RegisterController::class, 'view'])->name('register');
 Route::post('/register', [RegisterController::class, 'register']);
-Route::get('/', [UserController::class, 'profile']);
+Route::get('/profile', [UserController::class, 'profile'])->name('user.profile');
+Route::post('/logout', [UserController::class, 'logout'])->name('logout');
```
Comment (rafael-parayno) [inline src/routes/web.php]: Add new line

Comment (karljerico) [reply src/routes/web.php]: added new line sir, thank you!

File: src/app/Http/Controllers/LoginController.php [hunk 4]
```diff
@@ -3,11 +3,28 @@
...
+use App\Http\Requests\LoginRequest;
 
 class LoginController extends Controller
 {
     public function index(): View
     {
         return view('auth/login');
     }
+
+    public function auth(LoginRequest $request)
```
Comment (hipe01mark) [inline src/app/Http/Controllers/LoginController.php]: Add return type

File: src/app/Http/Requests/LoginRequest.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class LoginRequest extends FormRequest
+{
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
```
Comment (hipe01mark) [inline src/app/Http/Requests/LoginRequest.php]: Remove docblock.

File: src/app/Http/Controllers/LoginController.php (hunk 4 above)
Comment (karljerico) [reply src/app/Http/Controllers/LoginController.php]: Added return type Sir, Thank you!

File: src/app/Http/Requests/LoginRequest.php (hunk 5 above)
Comment (karljerico) [reply src/app/Http/Requests/LoginRequest.php]: Doblock is removed Sir, Thank you!

**Review by hipe01mark — DISMISSED** [review]
LGTM 🙇

File: src/app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -8,6 +8,12 @@ class UserController extends Controller
 {
     public function profile(): View
     {
-        return view('user.profile');
+        return view('user/profile');
     }
+
+    public function logout() {
```
Comment (bpocrafael) [inline src/app/Http/Controllers/UserController.php]: return type

**Review by bpocrafael — DISMISSED** [review]
LGTM 🐑 just one comment

**Review by hipe-francis — DISMISSED** [review]
LGTM 🚀

File: src/routes/web.php [hunk 7]
```diff
@@ -1,23 +1,13 @@
...
-
-
-Route::get('/', [LoginController::class, 'index'])->name('login');
+Route::get('/', [LoginController::class, 'index'])->name('login.auth');
+Route::post('/', [LoginController::class, 'auth']);
 Route::get('/register', [RegisterController::class, 'view'])->name('register');
 Route::post('/register', [RegisterController::class, 'register']);
-Route::get('/', [UserController::class, 'profile']);
+Route::get('/profile', [UserController::class, 'profile'])->name('user.profile');
+Route::post('/logout', [UserController::class, 'logout'])->name('logout');
```
Comment (hipe01mark) [inline src/routes/web.php]: Follow the route naming convention!

https://webdevetc.com/blog/laravel-naming-conventions/

Comment (karljerico) [reply src/routes/web.php]: Resolved this comment sir, Thankyou!

**Review by bpocrafael — DISMISSED** [review]
lgtm

File: src/app/Http/Controllers/RegisterController.php [hunk 8]
```diff
@@ -7,21 +7,20 @@
...
 
-    public function register(RegisterRequest $request): RedirectResponse
+    public function store(RegisterRequest $request): RedirectResponse
     {
         $incomingFields = $request->validated();
         $incomingFields['password'] = bcrypt($incomingFields['password']);
         $user = User::create($incomingFields);
         auth()->login($user);
-        return redirect('/');
+        return redirect('profile/show');
```
Comment (hipe-francis) [inline src/app/Http/Controllers/RegisterController.php]: You can use the declared name for this route.

File: src/app/Http/Controllers/UserController.php [hunk 9]
```diff
@@ -2,13 +2,20 @@
...
+    public function show(): View
     {
-        return view('user.profile');
+        return view('user/profile');
+    }
+
+    public function logout(): RedirectResponse
+    {
+        auth()->logout();
+        return redirect('/');
```
Comment (hipe-francis) [inline src/app/Http/Controllers/UserController.php]: Same goes here, if  you have a declared route name for '/' you should use that instead

File: src/routes/web.php [hunk 10]
```diff
@@ -1,24 +1,14 @@
...
-|
-*/
-
-
-Route::get('/', [LoginController::class, 'index'])->name('login');
-Route::get('/register', [RegisterController::class, 'view'])->name('register');
-Route::post('/register', [RegisterController::class, 'register']);
-Route::get('/profile', [UserController::class, 'profile'])->name('profile');
+Route::get('/', [LoginController::class, 'index'])->name('login.view');
+Route::post('/', [LoginController::class, 'auth'])->name('login,.auth');
```
Comment (hipe01mark) [inline src/routes/web.php]: Check the route name. is this working??

**Review by bpocrafael — DISMISSED** [review]

**Review by bpocrafael — APPROVED** [review]

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Pull request merged**